[DataLoader] Fix util ImportError

ejguan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#52459 [DataLoader] Fix util ImportError**

Differential Revision: [D26523941](https://our.internmc.facebook.com/intern/diff/D26523941)

ejguan committed:
[DataLoader] Fix util ImportError

[ghstack-poisoned]

glaringlee reviewed (approved):
lgtm

VitalyFedyunin commented on torch/utils/data/datapipes/utils/decoder.py:
```diff
@@ -182,8 +199,16 @@ def torch_video(key, data):
+    except ImportError as e:
+        try:
+            import pip  # type: ignore
+            pip.main(['install', '--user', 'torchvision'])
```
Please DO NOT install things on users behalf, throw error with instruction how to do it

ejguan replied:
It makes sense.

VitalyFedyunin reviewed (changes requested):
Not not auto-install things, it is not secure, might screw-up users setup etc etc

ejguan committed:
Update on "[DataLoader] Fix util ImportError"


Differential Revision: [D26523941](https://our.internmc.facebook.com/intern/diff/D26523941)

[ghstack-poisoned]

ejguan commented on torch/utils/data/datapipes/utils/decoder.py:
```diff
@@ -1,13 +1,15 @@
 import json
-import tempfile
-import io
+import numpy as np
```
`numpy` is one of the requirements for pytorch, so it should be fine for direct import

ejguan committed:
Update on "[DataLoader] Fix util ImportError"


Differential Revision: [D26523941](https://our.internmc.facebook.com/intern/diff/D26523941)

[ghstack-poisoned]

ejguan committed:
Update on "[DataLoader] Fix util ImportError"


Differential Revision: [D26523941](https://our.internmc.facebook.com/intern/diff/D26523941)

[ghstack-poisoned]

ejguan committed:
Update on "[DataLoader] Fix util ImportError"


Differential Revision: [D26523941](https://our.internmc.facebook.com/intern/diff/D26523941)

[ghstack-poisoned]
